packages/bridge/src/bridges/index.ts

```diff
@@ -0,0 +1,87 @@
+
+
+import { NetworkConfig } from '@arcadeum/provider'
+import { BigNumberish, providers } from 'ethers'
+
+export type MoveEstimate = {
+  crossTime: BigNumberish,
+  steps: BigNumberish
+}
+
+export type Move = {
+  transactionHash: string,
+  completeTx: providers.TransactionRequest[] | undefined,
+  isCompleted: boolean,
+  isPending: boolean,
+  fromChain: number,
+  toChain: number
+}
+
+export type MoveNative = Move & {
+  amount: BigNumberish
+}
+
+export type MoveERC20 = Move & {
+  token: string,
+  amount: BigNumberish
+}
+
+export type MoveERC721 = Move & {
+  token: string,
+  id: BigNumberish
+}
+
+export type MoveERC1155 = Move & {
+  token: string,
+  id: BigNumberish,
+  amount: BigNumberish
+}
+
+export function isERC20Move(move: Move): move is MoveERC20 {
+  const cand = move as MoveERC20 & MoveERC721
+  return cand.token !== undefined && cand.amount !== undefined && !cand.id
+}
+
+export function isERC721Move(move: Move): move is MoveERC721 {
+  const cand = move as MoveERC721 & MoveERC1155
+  return cand.token !== undefined && cand.id !== undefined && !cand.amount
+}
+
+export function isERC1155Move(move: Move): move is MoveERC721 {
+  const cand = move as MoveERC1155
+  return cand.token !== undefined && cand.id !== undefined && cand.amount !== undefined
+}
+
+export function isNativeMove(move: Move): move is MoveNative {
+  const cand = move as MoveERC20 & MoveERC721
+  return !isERC20Move(move) && cand.amount !== undefined && !cand.id
+}
+
+export interface Bridge {
```

Comment:
I really like the simplicity of the interface, but couldn't help myself trying to see if it could be simplified or where it could lead to some issues. The main problem I saw was if a certain bridge had other standards it supported that aren't mentioned here, updating the interface for all bridges could be a bit annoying. 

Perhaps an alternative approach could be to have the bridge on support 
```typescript
export interface Bridge {
  name(): string
  id(): string
  connect(networks: NetworkConfig[]): Bridge
  getMoves(wallet: string, from: providers.BlockTag, to: providers.BlockTag): Promise<Move[]>

  supports(assetInterface, ...): Promise<boolean>
  estimate(assetInterface, ...): Promise<MoveEstimate | undefined>
  move(assetInterface, ...): Promise<providers.TransactionRequest[] | undefined>
  complete(assetInterface, ...): Promise<providers.TransactionRequest[] | undefined>
}
```

And then the implementation could have a switch statement for support and to handle the various token standards, both existing ones and future ones. This would be future proof as if someone added a new token and a bridge that supported that new tokens, we wouldn't need to change the bridge interface and wouldn't need to adapt the other bridges as well since they would just naturally return false if you call `bridge.supports(newTokenInterface, ...)` that they didn't support.

What do you think?

Reply:
At first glance it sounds like a good idea, I will give it a try and see how it looks. I think it's better to keep the library easy-to-use over easy-to-implement, and I worry that having a single shared by all asset types could be confusing to use or be error-prone, more so if the dev is not using typescript.